libindy-crypto/docs/anoncreds-design.md

```diff
@@ -28,6 +28,17 @@ Initial version of anoncreds protocol was implemented as part of Indy SDK (https
 1. FFI C API will use OpenSSL style entities handling. Entities referenced will be represent as untyped pointers. Library will provide functions for entities allocation, manipulation and deallocation. 
 
 ## API 
+### API V2
+#### Reasons
+There are some significant aspects was missed in 1st generation. The new one was created to fix it:
+* Indy Crypto should have ability to work with large volume of Tails (can be larger rather RAM)
+* API should be well-mapped with main usecases with Indy Ledger.
+
+#### Changes
+* `RevocationRegistry` now will be created without full `Tails` in RAM as part of returned value.
```

Comment:
1. We should rename RevocationRegistry to RevocationRegistryDef
2. RevocationRegistryDef becomes immutable entity. We define 2 new entities that early were mutable part of RevocationRegistry:
- RevocationRegistryAcc - accumulator. 
- RevocationRegistryEntry - registry entry. Multiple entries can be combined to one. Each entry also contains accumulator value